staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go

```diff
@@ -709,51 +709,59 @@ func toJSON(b *testing.B, list *v1.PodList) []byte {
 	return out
 }
 
-func benchmarkSerializeObject(b *testing.B, payload []byte) {
-	input, output := len(payload), len(gzipContent(payload, defaultGzipContentEncodingLevel))
-	b.Logf("Payload size: %d, expected output size: %d, ratio: %.2f", input, output, float64(output)/float64(input))
-
+func benchmarkSerializeObject(b *testing.B, payload []byte, gzip bool) {
 	req := &http.Request{
-		Header: http.Header{
-			"Accept-Encoding": []string{"gzip"},
-		},
 		URL: &url.URL{Path: "/path"},
 	}
+	if gzip {
+		req.Header = http.Header{
+			"Accept-Encoding": []string{"gzip"},
+		}
+	}
+
 	featuregatetesting.SetFeatureGateDuringTest(b, utilfeature.DefaultFeatureGate, features.APIResponseCompression, true)
 
 	encoder := &fakeEncoder{
 		buf: payload,
 	}
 
 	b.ResetTimer()
+	responseBytesTotal := 0
 	for i := 0; i < b.N; i++ {
```

Comment:
Please consider using `b.Loop` to further improve this benchmark: https://pkg.go.dev/testing#B.Loop